### PR TITLE
Dala 7363 nuclear and gas data points for eu taxo non financial 2026

### PR DESCRIPTION
**Pull request by julianhoelzdfine**
# Pull Request \ Dala 7363 nuclear and gas data points for eu taxo non financial
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The GitHub Actions for the CI are green. This is enforced by GitHub. 
- [x] The PR has been peer-reviewed
  - [x] The code has been manually inspected by someone who did not implement the feature
  - [x] The changes have been inspected for possible breaking API changes (changed data models, endpoints, response codes, exception handling, ...). If there are any discuss them with the team.
  - [x] If this PR includes work on the frontend, at least one `@ts-nocheck` is removed. Additionally, there should not be any `@ts-nocheck` in files modified by this PR. If no `@ts-nocheck` are left: Celebrate :tada: :confetti_ball: type-safety and remove this entry. 
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] Make sure you understand the entire workflow from the user’s perspective, and test/review it accordingly. If anything is unclear, please reach out to the dev team.
- [x] At least one test exists testing the new feature
  - [ ] Make sure all newly added functionality (especially user facing) is tested
  - [ ] Make sure that new test files are included in a test container and actually run in the CI
- [x] At least 3 consecutive CI runs are successfully executed to ensure that there are no flaky tests.
- [x] Documentation is updated as required. If unsure whether or what to update, ask a fellow developer.
- [ ] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [ ] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [ ] A fresh clone of dataland.com is generated (see Wiki page on "OTC" for details)
  - [ ] The feature branch is deployed to clone with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on the clone server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team
  - [ ] The feature branch is deployed to one of the dev servers with `Reset non-user related Docker Volumes & Re-populate` turned on, and it's verified that the CD run is green
- [x] ELSE, the new version is deployed to one of the dev servers using this branch
  - [ ] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [ ] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on dev server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team
- [ ] Confirm that the latest version (use the /gitinfo endpoint) of the feature branch is deployed to one of the dev servers (no longer enforced by GitHub)
- [ ] Ensure that the release notes are written in a way that is understandable to a non-technical audience.

### CODE REVIEW

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/dataPoints/EuTaxonomyShareCalculationValidationTest.kt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+class EuTaxonomyShareCalculationValidationTest {
+    /**
+     * The following test checks if there appears an exception if the wrong number of inputs is entered
+     */
+    @Test
+    fun `check that EuTaxonomyShare rejects an empty input`() {
+        // Case 1: There are zero inputs
```
Comment (florian-hoeft) [inline dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/dataPoints/EuTaxonomyShareCalculationValidationTest.kt]: remove this comment

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/dataPoints/EuTaxonomyShareCalculationValidationTest.kt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import org.dataland.datalandbackend.utils.sourceFrameworksByType
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+class EuTaxonomyShareCalculationValidationTest {
+    /**
+     * The following test checks if there appears an exception if the wrong number of inputs is entered
+     */
+    @Test
+    fun `check that EuTaxonomyShare rejects an empty input`() {
```
Comment (florian-hoeft) [inline dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/dataPoints/EuTaxonomyShareCalculationValidationTest.kt:19]: This comment regards all 5 tests here that basically check when the EuTaxonomyShare rule throws an IllegalArgumentException. 

I noticed that tracing down the call path here for "applyTransformation", what actually happens in all of these tests is the following:
- applyTransformation is called with targetType = ACTIVITY_MERGE_RESULT_TYPE , which is just "resultType". 
- We land in convertEuTaxonomyShare
- In there, we call resolveEuTaxonomyShareRule(targetType)

resolveEuTaxonomyShareRule(targetType) will always throw an "IllegalArgumentException", because the ACTIVITY_MERGE_RESULT_TYPE is unknown. 

So these tests pass NOT because they check what they describe, but because an IllegalArgumentException is always thrown by the "resolveEuTaxonomyShareRule(inputType)" call. 

For example, the test "check that EuTaxonomyShare rejects an empty input" will fail because resolveEuTaxonomyShareRule(inputType) throws, not because we have an empty input, which would only later be verified. 

You would need to adjust the tests accordingly. For example by giving them a proper input type that passes. 
Also, if done, I would add one single test to check for resolveEuTaxonomyShareRule(inputType) with an unknown inputType

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/datapoints/DataPointConversionOperandsEuTaxonomy.kt [hunk 3]
```diff
@@ -110,24 +181,27 @@ internal inline fun <
...
 
     val nonAlignedInput =
         inputs.singleOrNull {
             getDataPointBaseTypeId(it.dataPointType, specs) == EuTaxonomyRulesConfig.NON_ALIGNED_ACTIVITIES_BASE_TYPE
         }
-            ?: throw IllegalArgumentException(
-                "Exactly one input of base type ${EuTaxonomyRulesConfig.NON_ALIGNED_ACTIVITIES_BASE_TYPE} " +
-                    "must be provided for the merge.",
-            )
+    if (nonAlignedInput == null) {
```
Comment (florian-hoeft) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/datapoints/DataPointConversionOperandsEuTaxonomy.kt]: My thoughts here: We check the nonAlignedInput and alignedInput SEQUENTIALLY here. 
This is important for the tests. If you create a test case (such as in EuTaxonomyShareCalculationValidationTest.kt) with two seperate alignedInputs, for example:
applyTransformation(
listOf(
createAlignedInput(listOf(alignedActivity())),
createAlignedInput(listOf(alignedActivity())),
),
ACTIVITY_MERGE_RESULT_TYPE, "EuTaxonomyShare", activityMergeSpecs, sourceFrameworksByType,
)

Then what actually throws here is that (nonAlignedInput == null). To actually tests "duplicated aligned inputs get rejected", the input would have to include one non-aligned input, and two aligned inputs. 

This needs to be taken into account when writing the tests

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/dataPoints/EuTaxonomyShareCalculationEligibleOrAlignedTest.kt [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+import org.dataland.datalandbackend.utils.assertBigDecimalEquals
+import org.dataland.datalandbackend.utils.createEligibleOrAlignedInput
+import org.dataland.datalandbackend.utils.eligibleOrAlignedActivity
+import org.dataland.datalandbackend.utils.shareSpecs
+import org.dataland.datalandbackend.utils.sourceFrameworksByType
+import org.dataland.datalandbackendutils.utils.JsonUtils.defaultObjectMapper
+import org.junit.jupiter.api.Test
+import java.math.BigDecimal
+
+class EuTaxonomyShareCalculationEligibleOrAlignedTest {
```
Comment (florian-hoeft) [inline dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/dataPoints/EuTaxonomyShareCalculationEligibleOrAlignedTest.kt:20]: The ticket mentions a test that verifies the behavior for one input with two activities who have an identical activityName. This is not covered. 

As implemented, two activities with the same activityName are treated exactly like two with different names: They get summed up. 

It is worth adding a test for this in my opinion.

**Review by florian-hoeft — COMMENTED** [review]
Summary of Findings: The 8 fields are correctly defined (right names, right sections — Revenue/CapEx only, correctly excluding OpEx — with NoUpload set), and CI is green. Calculation-logic test coverage for the new framework, old framework, and comment-generation is solid and well-targeted. However, EuTaxonomyShareCalculationValidationTest.kt has a confirmed bug: all 5 of its tests pass for the wrong reason — they throw due to an unrelated, mismatched target-type constant (ACTIVITY_MERGE_RESULT_TYPE) causing an earlier, unrelated IllegalArgumentException, rather than actually exercising the input-validation logic they're named after; this means the real argument-shape/duplicate-input validation is currently unverified by any test (details in the separately posted comment). Additionally, the ticket's own specified edge case — duplicate activity name within a single reported list — has no dedicated test anywhere, though by code inspection the summation logic appears to handle it correctly "by construction." The missing totalShare/"Incomplete" quality logic was confirmed with the author as an intentional scope decision, not a gap. Frontend Vue changes and full deep-dive correctness review of the pre-existing Kotlin conversion code were not completed in this pass due to time constraints.


Also: The CI is not green yet, because there are some Sonar Code Smells

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/datapoints/DataPointConversionOperandsEuTaxonomy.kt (hunk 3 above)
Comment (DominikTschimmel-d-fine) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/datapoints/DataPointConversionOperandsEuTaxonomy.kt]: Unfortunately, we cannot test with 3 inputs. They get rejected earlier (dispatch based on input number). So I renamed the respective test to indicate that it checks for missing non-aligned input, not for duplicates.

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/dataPoints/EuTaxonomyShareCalculationValidationTest.kt (hunk 1 above)
Comment (DominikTschimmel-d-fine) [reply dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/dataPoints/EuTaxonomyShareCalculationValidationTest.kt]: Done

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/dataPoints/EuTaxonomyShareCalculationValidationTest.kt (hunk 2 above)
Comment (DominikTschimmel-d-fine) [reply dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/dataPoints/EuTaxonomyShareCalculationValidationTest.kt:19]: Now using a proper target type and added an additional test with `ACTIVITY_MERGE_RESULT_TYPE` as "unknown target type" test

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/dataPoints/EuTaxonomyShareCalculationEligibleOrAlignedTest.kt (hunk 4 above)
Comment (DominikTschimmel-d-fine) [reply dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/dataPoints/EuTaxonomyShareCalculationEligibleOrAlignedTest.kt:20]: Done

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/dataPoints/EuTaxonomyShareCalculationValidationTest.kt (hunk 1 above)
Comment (florian-hoeft) [reply dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/dataPoints/EuTaxonomyShareCalculationValidationTest.kt]: resolved

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/dataPoints/EuTaxonomyShareCalculationValidationTest.kt (hunk 2 above)
Comment (florian-hoeft) [reply dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/dataPoints/EuTaxonomyShareCalculationValidationTest.kt:19]: Thanks. Looks good to me. Resolved.

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/datapoints/DataPointConversionOperandsEuTaxonomy.kt (hunk 3 above)
Comment (florian-hoeft) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/datapoints/DataPointConversionOperandsEuTaxonomy.kt]: You are right. Thanks. Resolved.
